docs/elasticsearch/ux.md

```diff
@@ -0,0 +1,1240 @@
+    "simple_phrase": [
+      {
+        "text": "That is the qution",
+        "offset": 0,
+        "length": 18,
+        "options": [
+          {
+            "text": "that is the question",
+            "score": 0.0015543294
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+
+## Pagination
+
+Use the `from` and `size` parameters to return results to your users one page at time.
+
+The `from` parameter indicates the document number that you want to start showing the results from. The `size` parameter refers to the number of results that you want to show. The `from` and `size` parameters together act as a sliding window that moves through the results.
+
+For example, if the value of `size` is 10 and the value of `from` is 0, you see the first 10 results. If you change the value of `from` to 10, you see the next 10 results (because the results are zero-indexed). So, if you want to see results starting from result 11, `from` must be 10.
+
+```json
+GET shakespeare/_search
+{
+  "from": 0,
+  "size": 10,
+  "query": {
+    "match": {
+      "play_name": "Hamlet"
+    }
+  }
+}
+```
+
+```json
+Search Results
+------------- from: 0, size: 10
+1
+2
+3
+4
+5
+6
+7
+8
+9
+10
+------------- from: 10, size: 10
+11
+12
+13
+14
+15
+16
+17
+18
+19
+20
+```
+
+To calculate the `from` parameter relative to the page number:
+
+```json
+from = size * (page_number - 1)
+```
+
+Each time the user clicks the next page of the results, your application must make the same search query with an incremented `from` value.
+
+You could also specify the `from` and `size` parameters in the search URI:
+
+```json
+GET shakespeare/_search?from=0&size=10
+```
+
+If you only specify the `size` parameter, the `from` parameter defaults to 0.
+
+Querying for pages deep in your results could have a significant performance impact, so this approach is limited to 10000 results.
+
+The `from` and `size` parameters are completely stateless, so the results are always based on the latest available data.
+This could cause inconsistent pagination.
+For example, assume a user stays on the first page of the results for a minute and then navigates to the second page; in that time, a new document is indexed in the background which is relevant enough to show up on the first page. In this scenario, the last result of the first page is pushed to the second page, so the user ends up seeing a result on the second page that they already saw on the first page.
+
+Use the `scroll` operation for consistent pagination. With the `scroll` operation, a search context is kept open for a certain period of time. Any data changes do not affect the results during this time.
+
+## Scroll
+
+The `from` and `size` parameters allow you to paginate your search results, but with a limit of 10000 results at a time. If you need to request massive volumes of data for say a machine learning job, use the `scroll` operation instead. The `scroll` operation allows you to request an unlimited number of results.
```

Comment:
data from, for example, a machine learning job, use the

Reply:
Done.